### PR TITLE
new function shorten

### PR DESCRIPTION
**Pull request by Mrx-dev**
shorten function let you short your URL with tinyurl !

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -53,6 +53,8 @@ export function boostImage(url: string): Promise<string>;
 
 export function start(client: Client, boostRoleID: Snowflake): Promise<void>;
 
+export function shorten(url: string): Promise<String>;
```
Comment (reinacchi) [inline index.d.ts]:
```suggestion
export function shorten(url: string): Promise<string>;
```

File: functions/shorten.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
+const Discord = require("discord.js");
+
+/**
+* Create a short url for a Discord.js Bot
+ * @param {String} url the url
+ * @returns {Promise<String>} returns the short url
+ */
```
Comment (reinacchi) [inline functions/shorten.js]:
```suggestion
/**
* Create a short url for a Discord.js Bot
 * @param {String} url The URL to be shorten
 * @returns {Promise<String>} 
 */
```

File: functions/shorten.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+            require.resolve("tinyurl");
+        } catch (e) {
+            throw new Error("[UltraX] => Cannot find module 'tinyurl' Please do ' npm i tinyurl@latest'");
+        }
+        if (!url) throw new SyntaxError(`shorten =>  "No url provided"`);
+        if (typeof url != "string") throw new SyntaxError(`shorten => "the url it should be string"`);
+        if (!is_url) throw new ReferenceError(`shorten => the url must starts with http:// or https://`);
+        TinyURL.shorten(url, function (res, error) {
+            if (error || res == "Error") {
+                reject("shorten => Somthing Goes Wrong");
```
Comment (reinacchi) [inline functions/shorten.js]:
```suggestion
            throw new Error("[UltraX] => Cannot find module 'tinyurl' Please run 'npm i tinyurl@latest'");
        }
        if (!url) throw new SyntaxError(`Shorten =>  "No url provided"`);
        if (typeof url != "string") throw new SyntaxError(`Shorten => "The url should be type string"`);
        if (!is_url) throw new ReferenceError(`Shorten => The url must starts with http:// or https://`);
        TinyURL.shorten(url, function (res, error) {
            if (error || res == "Error") {
                reject("Shorten => Somthing Goes Wrong");
```

**Review by reinacchi — CHANGES_REQUESTED** [review]

File: functions/shorten.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
+const Discord = require("discord.js");
```
Comment (reinacchi) [inline functions/shorten.js:1]: you can remove this line since it's not declared anywhere

**Review by reinacchi — CHANGES_REQUESTED** [review]
I missed this one

File: functions/shorten.js (hunk 4 above)
Comment (MrxDevv) [reply functions/shorten.js:1]: cool

File: functions/shorten.js (hunk 3 above)
Comment (RaZeSloth) [reply functions/shorten.js]: I was about to comment the grammar.

File: functions/shorten.js (hunk 4 above)
Comment (reinacchi) [reply functions/shorten.js:1]: ~~doubt you used GitHub before~~ umm you can remove that line since it's not necessary to be there

Comment (MrxDevv) [reply functions/shorten.js:1]: yeah i don't use github a lot this is my first contribute :3 

sry for forget a lot of things i'm not used to it :)

Comment (reinacchi) [reply functions/shorten.js:1]: > you can remove this line since it's not declared anywhere

waiting for this...

**Comment by RaZeSloth** [comment]
Due to @KarimX32 request, this RP will be closed due to him implementing it in his [api](https://api.ultrax-yt.com/)

**Pull request closed without merging**